docs/src/rules/no-restricted-imports.md

```diff
@@ -282,6 +282,107 @@ import { AllowedObject } from "foo";
 
 :::
 
+#### allowTypeImports (TypeScript only)
+
+Whether to allow [Type-Only Imports](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export) for a path. This includes type-only `export` statements, as they are equivalent to re-exporting an `import`. Default: `false`.
+
+```json
+"no-restricted-imports": ["error", {
+  "paths": [{
+    "name": "import-foo",
+    "allowTypeImports": true,
+  }]
+}]
+```
+
+Examples of **incorrect** code for `allowTypeImports` in `paths`:
+
+::: incorrect { "sourceType": "module" }
+
+```ts
+/*eslint no-restricted-imports: ["error", { paths: [{
+    name: "import-foo",
+    allowTypeImports: true,
+    message: "Please use only type-only imports from 'import-foo'."
+}]}]*/
+
+import foo from 'import-foo';
+export { Foo } from 'import-foo';
+
+import baz from 'import-baz';
+export { Baz } from 'import-baz';
+```
+
+:::
+
+Examples of **correct** code for `allowTypeImports` in `paths`:
+
+::: correct { "sourceType": "module" }
+
+```ts
+/*eslint no-restricted-imports: ["error", { paths: [{
+    name: "foo",
+    allowTypeImports: true,
+    message: "Please use only type-only imports from 'import-foo'."
+}]}]*/
+
+import { foo } from 'other-module';
```

Comment:
```suggestion
```

I think this isn't necessary for this example, and it might be better for understanding to just match the incorrect code example above (just with the `type` keyword added).